documenteer/sphinxconfig/utils.py

```diff
@@ -29,10 +29,152 @@ def read_git_branch():
             return ''
 
 
-def read_git_commit_timestamp():
-    """Obtain the timestamp from the current git commit."""
-    repo = git.repo.base.Repo(search_parent_directories=True)
-    return repo.head.commit.committed_datetime
+def read_git_commit_timestamp(repo_path=None):
+    """Obtain the timestamp from the current head commit of a Git repository.
+
+    Parameters
+    ----------
+    repo_path : `str`, optional
+        Path to the Git repository. Leave as `None` to use the current working
+        directory.
+
+    Returns
+    -------
+    commit_timestamp : `datetime.datetime`
+        The datetime of the head commit.
+    """
+    repo = git.repo.base.Repo(path=repo_path, search_parent_directories=True)
+    head_commit = repo.head.commit
+    return head_commit.committed_datetime
+
+
+def read_git_commit_timestamp_for_file(filepath, repo_path=None):
+    """Obtain the timestamp for the most recent commit to a given file in a
+    Git repository.
+
+    Parameters
+    ----------
+    filepath : `str`, optional
+        Repository-relative path for a file.
+    repo_path : `str`, optional
+        Path to the Git repository. Leave as `None` to use the current working
+        directory.
+
+    Returns
+    -------
+    commit_timestamp : `datetime.datetime`
+        The datetime of a the most recent commit to the given file.
+
+    Raises
+    ------
+    IOError
+        Raised if the ``filepath`` is set, but does not exist in the Git
+        repository.
+    """
+    repo = git.repo.base.Repo(path=repo_path, search_parent_directories=True)
+    head_commit = repo.head.commit
+
+    # most recent commit datetime of the given file
+    for commit in head_commit.iter_parents(filepath):
+        return commit.committed_datetime
+
+    # Only get here if git could not find the file path in the history
+    raise IOError('File {0} not found'.format(filepath))
+
+
+def get_filepaths_with_extension(extname, root_dir='.'):
+    """Get relative filepaths of files in a directory, and sub-directories,
+    with the given extension.
+
+    Parameters
+    ----------
+    extname : `str`
+        Extension name (e.g. 'txt', 'rst').
+    root_dir : 'str`, optional
+        Root directory. Current working directory by default.
+
+    Return
+    ------
+    filepaths : `list` of `str`
+        File paths, relative to ``root_dir``, with the given extension.
+    """
+    if not extname.startswith('.'):
+        extname = '.' + extname
+
+    root_dir = os.path.abspath(root_dir)
+
+    selected_filenames = []
+    for dirname, sub_dirnames, filenames in os.walk(root_dir):
+        for filename in filenames:
+            if os.path.splitext(filename)[-1].lower() == extname:
```

Comment:
Why lower case here? If the file is `.TXT` and `extname` is `.TXT` then this will never match. If you are being case-insensitive you really should say so in the description and you should ensure that `extname` is lower cased.